OpenRA.Mods.Common/Traits/Buildings/Building.cs

```diff
@@ -173,6 +172,15 @@ public Actor FindBaseProvider(World world, Player p, CPos topLeft)
 
 		public virtual bool IsCloseEnoughToBase(World world, Player p, string buildingName, CPos topLeft)
 		{
+			Func<Actor, bool> hasAdjanentType = a =>
```

Comment:
Typo here.

Reply:
Also you should move this to be defied closer to the loop where it's used. At the moment if the two following conditions bail out this goes unused.

Reply:
Would it be worth pulling this out into a proper method to avoid unnecessary object creation?

Reply:
Yes.